packages/components/pie-modal/src/index.ts

```diff
@@ -31,17 +36,78 @@ export class PieModal extends RtlMixin(LitElement) {
         this.addEventListener('click', (event) => this._handleDialogLightDismiss(event));
     }
 
+    firstUpdated (changedProperties: DependentMap<ModalProps>) : void {
+        this._handleModalOpenStateOnFirstRender(changedProperties);
+    }
+
+    updated (changedProperties: DependentMap<ModalProps>) : void {
+        this._handleModalOpenStateChanged(changedProperties);
+    }
+
+    private _handleModalOpened () : void {
+        this._disableScrolling();
+        //  We require this because toggling the prop `isOpen` itself won't
+        //  allow the dialog to open in the correct way (with the default background),
+        //  the method `showModal()` needs to be invoked.
+        this._dialog?.showModal();
+    }
+
+    private _handleModalClosed () : void {
+        this._enableScrolling();
+        //  Closes the native dialog element
+        this._dialog?.close();
+    }
+
+    connectedCallback () : void {
+        super.connectedCallback();
+        document.addEventListener(ON_MODAL_OPEN_EVENT, this._handleModalOpened.bind(this));
+        document.addEventListener(ON_MODAL_CLOSE_EVENT, this._handleModalClosed.bind(this));
+    }
+
+    disconnectedCallback () : void {
+        document.removeEventListener(ON_MODAL_OPEN_EVENT, this._handleModalOpened.bind(this));
+        document.removeEventListener(ON_MODAL_CLOSE_EVENT, this._handleModalClosed.bind(this));
+        super.disconnectedCallback();
+    }
+
+    // Handles the value of the isOpen property on first render of the component
+    private _handleModalOpenStateOnFirstRender (changedProperties: DependentMap<ModalProps>) : void {
+        // This ensures if the modal is open on first render, the scroll lock and backdrop are applied
+        if (changedProperties.has('isOpen')) {
+            const previousValue = changedProperties.get('isOpen') as boolean;
+            if (previousValue === undefined && this.isOpen) {
+                this._dispatchModalOpenEvent();
+            }
+        }
+    }
+
+    // Handles changes to the modal isOpen property by dispatching any appropriate events
+    private _handleModalOpenStateChanged (changedProperties: DependentMap<ModalProps>) : void {
+        if (changedProperties.has('isOpen')) {
+            const previousValue = changedProperties.get('isOpen') as boolean;
```

Comment:
I still don't think we need to assert here but happy for you to leave it in. The method of the get will automatically return a boolean value.

Reply:
It won't automatically return a boolean, it will be bool or undefined. However the `as boolean` isn't needed and I'm pushing an update to remove that now :)